script/eval_lc_readiness.py

```
@@ -0,0 +1,614 @@
+def gql(query, variables, user):
+    body = json.dumps({"query": query, "variables": variables}).encode()
+    req = urllib.request.Request(
+        GRAPHQL,
+        data=body,
+        headers={
+            "Content-Type": "application/json",
+            "User-Agent": UA,
+            "Referer": f"https://leetcode.com/u/{user}/",
+        },
+    )
+    with urllib.request.urlopen(req, timeout=30) as r:
```

Ruff (0.16.2)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)